docs/reusing-typechecking-results.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Reusing typechecking results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing typechecking results between compiler runs
+
+Caching and reusing compilation results between compiler runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+## Motivation
+
+- **Performance**. Things like recomputing type information for large projects are time-consuming. By caching the results, subsequent compilation can skip some steps, leading to faster builds.
+
+- **Better IDE cold start**. IDEs can provide faster IntelliSense, code navigation, and other features if they can quickly access cached compilation information.
```

Comment:
Can you please expand more here? E.g. how caching can benefit different types of projects? Small vs large ones, tightly couples vs linear dependency graph, etc.

Reply:
Added some clarifications.